### PR TITLE
fix: pass sent timestamp in binance snapshot

### PR DESCRIPTION
**Pull request by shaunc**
### Description of code - what bug does this fix / what feature does this add?

- [ ] - Tested
- [ ] - Changelog updated
- [ ] - Tests run and pass
- [ ] - Flake8 run and all errors/warnings resolved
- [ ] - Contributors file updated (optional)

Unfortunately, I had trouble getting the tests running locally. Also didn't know where exactly test should go. However, this fix is very small & localized.

### CODE REVIEW

**Comment by bmoscon** [comment]
You'll need to test that confirm that this doesnt break any of the other binance exchanges since they use a shared codebase

File: cryptofeed/exchanges/binance.py [hunk 1]
```diff
@@ -285,7 +285,7 @@ async def _snapshot(self, pair: str) -> None:
         std_pair = self.exchange_symbol_to_std_symbol(pair)
         self.last_update_id[std_pair] = resp['lastUpdateId']
         self._l2_book[std_pair] = OrderBook(self.id, std_pair, max_depth=self.max_depth, bids={Decimal(u[0]): Decimal(u[1]) for u in resp['bids']}, asks={Decimal(u[0]): Decimal(u[1]) for u in resp['asks']})
-        await self.book_callback(L2_BOOK, self._l2_book[std_pair], time.time(), timestamp=None, raw=resp, sequence_number=self.last_update_id[std_pair])
+        await self.book_callback(L2_BOOK, self._l2_book[std_pair], time.time(), timestamp=self.timestamp_normalize(resp.get('E')), raw=resp, sequence_number=self.last_update_id[std_pair])
```
Comment (bmoscon) [inline cryptofeed/exchanges/binance.py:288]: this fails for some variations of binance, so please fix and re-test and then I will merge

**Review by bmoscon — CHANGES_REQUESTED** [review]

File: cryptofeed/exchanges/binance.py (hunk 1 above)
Comment (shaunc) [reply cryptofeed/exchanges/binance.py:288]: Right -- I can make it fail myself now ... will try to get a fix in later today. Thanks!

File: cryptofeed/exchanges/binance.py [hunk 2]
```diff
@@ -279,13 +279,19 @@ async def _snapshot(self, pair: str) -> None:
                     max_depth = d
                     break
 
+        req_time = time.time()
         resp = await self.http_conn.read(self.rest_endpoints[0].route('l2book', self.sandbox).format(pair, max_depth))
         resp = json.loads(resp, parse_float=Decimal)
+        if 'E' not in resp:
+            resp_time = time.time()
+            timestamp = req_time + (resp_time - req_time) / 2
```
Comment (bmoscon) [inline cryptofeed/exchanges/binance.py:287]: the timestamp field in the data object must always come from the exchange. you cannot generate a timestamp and put it in there. if the exchange does not provide a timestamp, it should be none, like the code currently is

**Review by bmoscon — CHANGES_REQUESTED** [review]

File: cryptofeed/exchanges/binance.py (hunk 2 above)
Comment (shaunc) [reply cryptofeed/exchanges/binance.py]: Ok -- resubmitted.

**Pull request merged**